File: frontend/src/types/manutencao.ts
/** TypeScript interfaces for Manutencao entity matching Pydantic schemas. */

export interface ManutencaoResponse {
  id: number;
  drone_id: number;
  tipo: string;
  descricao: string | null;
  data_manutencao: string;
  proxima_manutencao: string | null;
  horas_na_data: number | null;
  created_at: string;
  updated_at: string;
}

export interface ManutencaoCreate {
  drone_id: number;
  tipo: string;
  descricao?: string | null;
  data_manutencao: string;
  proxima_manutencao?: string | null;
  horas_na_data?: number | null;
}

export interface ManutencaoUpdate {
  tipo?: string | null;
  descricao?: string | null;
  data_manutencao?: string | null;
  proxima_manutencao?: string | null;
  horas_na_data?: number | null;
}

File: app/schemas/manutencao.py
"""Pydantic schemas for Manutencao entity."""

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class ManutencaoBase(BaseModel):
    drone_id: int
    tipo: str = Field(..., max_length=100)
    descricao: str | None = None
    data_manutencao: date
    proxima_manutencao: date | None = None
    horas_na_data: Decimal | None = Field(None, ge=0)


class ManutencaoCreate(ManutencaoBase):
    pass


class ManutencaoUpdate(BaseModel):
    tipo: str | None = Field(None, max_length=100)
    descricao: str | None = None
    data_manutencao: date | None = None
    proxima_manutencao: date | None = None
    horas_na_data: Decimal | None = Field(None, ge=0)


class ManutencaoResponse(ManutencaoBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
